src/main/java/org/apache/commons/pool2/impl/EvictionTimer.java

```diff
@@ -46,6 +51,9 @@
     /** Executor instance */
     private static ScheduledThreadPoolExecutor executor; //@GuardedBy("EvictionTimer.class")
 
+    /** Keys are weak references to tasks, values are runners managed by executor. */
+    private static final HashMap<WeakReference<Runnable>, WeakRunner> taskMap = new HashMap<>(); // @GuardedBy("EvictionTimer.class")

```

Comment:
Is a ConcurrentHashMap needed here?

Reply:
No, access is only inside sync blocks.  (see GuardedBy annotation).